### PR TITLE
Theme index enhancment support

### PR DESCRIPTION
**Pull request by mdaffin**
Adds support for loading index.js in themes.

This works in the same way that `.vuepress/enhanceApp.js` works but for themes instead, allowing theme designers to extend the vue app in more useful ways.

This should help with creating custom themes for things like blogging sites like requested in #36

### CODE REVIEW

File: lib/prepare.js [hunk 1]
```diff
@@ -62,6 +62,15 @@ if (!Object.assign) Object.assign = require('object-assign')`
...
+  // 7. handle the theme index.js
+  const hasThemeIndexPath = fs.existsSync(options.themeApp)
+  await writeTemp(
+    'themeApp.js',
+    hasThemeIndexPath
+      ? `export { default } from ${JSON.stringify(options.themeApp)}`
+      : `export default function () {}`
+  )
+
   return options
```
Comment (ulivz) [inline lib/prepare.js]: `themeApp ` and `enhanceApp` are the same except  `hasEnhancePath` and `hasThemeIndexPath`, which can be extracted.

Comment (mdaffin) [reply lib/prepare.js]: Not exactly the same, they are two proxy files to two different js files that may or may not exist. They will default to an empty function if they don't exist but are there to load in two different files `<theme>/index.js` and `.vuepress/enchanceApp.js` respectively into the main vue app.

Comment (ulivz) [reply lib/prepare.js]: hmmm... you misunderstand my comment, I know WHY and HOW you do that...
I just want you to refactor your code like this:
```js
  async function writeEnhanceTemp (destName, srcPath, isEnhanceExist) {
    await writeTemp(
      destName,
      isEnhanceExist
        ? `export { default } from ${JSON.stringify(srcPath)}`
        : `export default function () {}`
    )
  }

  // 6. handle enhanceApp.js
  const enhanceAppPath = path.resolve(sourceDir, '.vuepress/enhanceApp.js')
  writeEnhanceTemp('enhanceApp.js', enhanceAppPath, fs.existsSync(enhanceAppPath))

  // 7. handle the theme index.js
  writeEnhanceTemp('themeApp.js', options.themeApp, fs.existsSync(options.themeApp))
```

Comment (mdaffin) [reply lib/prepare.js]: Ah, ok, done

**Comment by mdaffin** [comment]
Before releasing this we should consider if we want to stick with `index.js` or use something else as we might want to use `index.js` for theme plugins as in #196

**Pull request merged**